transport_ssh.rb

```diff
@@ -0,0 +1,65 @@
+require 'bolt/transport/base'
+require 'bolt/transport/sudoable'
+require 'bolt/transport/ssh/connection'
+require 'concurrent/atomic/atomic_reference'
+require 'concurrent/promises'
+
+class BoltSSH < Bolt::Transport::SSH::Connection
+  # override execute so it streams output and returns the exit code
+  def execute(command, stdin: nil)
```

Comment:
Some of this code seems copy-pasted from https://github.com/puppetlabs/bolt/blob/master/lib/bolt/transport/ssh/connection.rb#L224 ? If so, might be helpful to link that as a reference.

Reply:
You know, given all the code here, maybe breaking consistency isn't so bad. If I was reading this code, I'd be happy seeing something like
```
if supports_ssh
  // Implementing exec for SSH sucks, so we're just going to use Wash's
  // transport package to do it for us. Unfortunately no support exists for
  // WinRM, so we'll have to implement our own version in exec.
  transport :ssh, ...
end
```

Reply:
It's not that much code, and it is pretty much copied from Bolt with minor changes.

Reply:
Ok, fair enough.